### ISSUE TITLE
[JavaScript] ES6: colons in template strings for computed object keys

### ISSUE THREAD

**Issue by Elberet** [sublimehq/Packages#880]
ES6 permits computed keys in object literals:
```javascript
let index = 0;
let obj = {
  [`key${index++}`]: 'zero',
  [`key${index++}`]: 'one'
};
// equivalent:
obj[`key${index++}`] = 'two';
```

Sublime's JS syntax highlighting breaks when such an object key contains a template string which in turn contains a colon:
![image](https://cloud.githubusercontent.com/assets/361662/24697049/03831ec6-19ec-11e7-9ae6-b87480ab9e44.png)

**Comment by michaelblyons** [comment]
It's bugging me that your code snippet is not the same as your screenshot<sup>1</sup>. I'm guessing that the snippet is a typo, but could you verify that.

It's also possible that I don't understand the ES6 being demonstrated.

<sup>1</sup>The snippet is missing the second backtick each time and contains neither a `-` nor a `:` between `key` and `$`.

**Comment by wbond** [comment]
I changed the snippet to have what I understand to be valid syntax:

```js
let index = 0;
let obj = {
  [`key${index++}`]: 'zero',
  [`key${index++}`]: 'one'
};
// equivalent:
obj[`key${index++}`] = 'two';
```

This seems to highlight properly with the latest syntax.

**Comment by wbond** [comment]
This also highlights properly:

```js
let anotherObj = {
  [`key-${index++}`]: getStyles('hover'),
  [`key-${index++}`]: getStyles('hover')
};

let anotherObj = {
  [`key:${index++}`]: getStyles('hover'),
  [`key:${index++}`]: getStyles('hover')
};
```

**Comment by Elberet** [comment]
@michaelblyons Whoops, that's what you get for copy&pasting. Sorry, I fixed it.

I also double-checked this on my machine at home, and lo and behold, the highlighting is fine here. Which I find strange, because I'm pretty sure I tested this "problem" with a stock Sublime installation before recreating the screenshot with my usual set of packages...

Oh well. Thanks anyway!

**Comment by Elberet** [comment]
Okay, I didn't just imagine that I had tested on a clean Sublime. Once again, on my workplace laptop, using a completely clean install of portable Sublime Text 3 build 3126 running on Win 7 Enterprise x64, the problem persists.

It did, however, go away after I installed the current version of the JavaScript syntax package from Github.

Thus, apparently, this is a bug that has already been fixed in the current version of the package on Github, while both the installer as well as the portable archive from the website contain older versions.

Which begs the question... I don't recall installing the Git version on my home machine; so how did I get the newer JS package on that machine? :sweat_smile:
